src/Conversion/ONNXToTorch/NN/SqueezeOp.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------- SqueezeOp.cpp - ONNX Op Transform ------------------===//
+//
+// Copyright 2022, Helprack LLC.
+//
+// =============================================================================
+//
+// This file lowers squeeze operator from onnx to torch using a template for
+// v11 and v13.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToTorch/ONNXToTorchCommon.hpp"
+#include "src/Conversion/ONNXToTorch/NN/CommonUtils.h"
+
+#include <cstdint>
+#include <set>
+#include <vector>
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+
+std::vector<int> toUniqueAndNonNegative(std::vector<int> axes) {
+  std::set<int> axesSet(axes.begin(), axes.end());
+  std::vector<int> axesNonNeg;
+
+  for (auto x : axesSet) {
+    // positive integers are added as it
+    // negative integers are normarlized to positive
+    axesNonNeg.push_back((x > 0) ? x : (x + axesSet.size()));
+  }
+  return axesNonNeg;
+}
+
+std::vector<int> getSortedWithNonNegativeAxes(mlir::ArrayAttr axesRaw) {
+  auto axesVector = toVector(axesRaw);
+  auto axesNonNegative = toUniqueAndNonNegative(axesVector);
+  auto axesSorted = axesNonNegative;
+
+  std::sort(axesSorted.begin(), axesSorted.end());
+
+  return axesSorted;
+}
+
+std::vector<int> getAxes(ONNXSqueezeOp squeezeOp) {
+  auto builder = mlir::Builder(squeezeOp.getContext());
+  auto axesConstOp = getONNXConstantOp(squeezeOp.axes());
+  auto axesAttr = createArrayAttrFromConstantOp(builder, axesConstOp);
+  return getSortedWithNonNegativeAxes(axesAttr);
+}
+
+std::vector<int> getAxes(ONNXSqueezeV11Op squeezeOp) {
```

Comment:
Please add some explanation regarding differences between `ONNXSqueezeV11Op` and `ONNXSqueezeOp`. When did you incur them? Is it related to different op sets?

Reply:
Thanks for looking into it!